precompiles/gov/integration_test.go

```diff
@@ -658,5 +659,77 @@ var _ = Describe("Calling governance precompile from EOA", func() {
 				Expect(out.Proposals).To(HaveLen(1))
 			})
 		})
+
+		Context("params query", func() {
+			var (
+				err                   error
+				callsData             CallsData
+				govCallerContractAddr common.Address
+				govCallerContract     evmtypes.CompiledContract
+			)
+
+			BeforeEach(func() {
+				// Setting gas tip cap to zero to have zero gas price.
+				txArgs.GasTipCap = new(big.Int).SetInt64(0)
+
+				govCallerContract, err = testdata.LoadGovCallerContract()
+				Expect(err).ToNot(HaveOccurred(), "failed to load GovCaller contract")
+
+				govCallerContractAddr, err = s.factory.DeployContract(
+					s.keyring.GetPrivKey(0),
+					evmtypes.EvmTxArgs{}, // NOTE: passing empty struct to use default values
+					factory.ContractDeploymentData{
+						Contract: govCallerContract,
+					},
+				)
+				Expect(err).ToNot(HaveOccurred(), "failed to deploy gov caller contract")
+				Expect(s.network.NextBlock()).ToNot(HaveOccurred(), "error on NextBlock")
+
+				callsData = CallsData{
+					precompileAddr: s.precompile.Address(),
+					precompileABI:  s.precompile.ABI,
+
+					precompileCallerAddr: govCallerContractAddr,
+					precompileCallerABI:  govCallerContract.ABI,
+				}
+			})
+
+			DescribeTable("should return all params", func(callType callType) {
+				txArgs, callArgs = callsData.getTxAndCallArgs(callArgs, txArgs, callType, "")
+
+				switch callType {
+				case directCall:
+					callArgs.MethodName = gov.GetParamsMethod
+				case contractCall:
+					callArgs.MethodName = "getParams"
+				}
+
+				_, ethRes, err := s.factory.CallContractAndCheckLogs(
+					s.keyring.GetPrivKey(0),
+					txArgs,
+					callArgs,
+					passCheck,
+				)
+				Expect(err).To(BeNil())
+
+				var output struct {
+					Params gov.ParamsOutput `json:"params"`
+				}
+				err = s.precompile.UnpackIntoInterface(&output, gov.GetParamsMethod, ethRes.Ret)
+				Expect(err).To(BeNil())
+
+				// Verify default params
+				Expect(output.Params.VotingPeriod).NotTo(BeNil())
+				Expect(output.Params.MaxDepositPeriod).NotTo(BeNil())
+				Expect(output.Params.MinDeposit).NotTo(BeNil())
+				Expect(output.Params.Quorum).NotTo(BeNil())
+				Expect(output.Params.Threshold).NotTo(BeNil())
+				Expect(output.Params.VetoThreshold).NotTo(BeNil())
+				Expect(output.Params.VotingPeriod).NotTo(BeNil())
```

Comment:
probably would be better to query the params using the gRPC and then comparing results; not just evaluate against `nil`